lesson8-UnitTesting/Sudoku/src/Sudoku.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Sudoku
+{
+    public class Sudoku
+    {
+        private static readonly HashSet<int> ValidSet = new HashSet<int>{1, 2, 3, 4, 5, 6, 7, 8, 9};
+
+        public static bool IsValid(int[,] solution)
+        {
+            if(solution == null)
+                throw new ArgumentNullException();
+            if(solution.GetLength(0) != solution.GetLength(1))
```

Comment:
For me this check is redundant as you can't two-dimensional arrays (`int[,]`) with different lines length, unlike in array of arrays (`int[][]`).

Reply:
deleted

Reply:
but what about this case:
`        [Test]
        public void ThrowsArgumentExceptionIfArrayIsNotSquare()
        {
            var invalidSolution = new int[2,3];
            Assert.Throws<ArgumentException>(() => Sudoku.IsValid(invalidSolution));
        }
`

Reply:
Ok, I mixed up the purpose of GetLength method.